### DISCUSSION TITLE
1.78

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#1979] [General]
## Changes

- Added support for **Beam Hash III** on Nvidia RTX 4000 and AMD RX 7000 generation GPUs
- Added support for **Heavyhash-Karlsen** for Nvidia Pascal and newer and AMD RX 6000 and newer GPUs. Use --algorithm KARLSEN to mine it, fee is 1%
- Added multiple **dual mining** options for EthashB3 algorithm (Rethereum). The algorithm can now be used with --dualmode to be set to ALEPHDUAL (Blake3 - Alephium), FISHDUAL (Blake3 - Ironfish), RXDDUAL (Sha512 / 256D, Radiant) or KARLSENDUAL (HeavyHash-Karlsen). Fee rule for dual mining: its the same as with the individual algorithms, so if EthB3 takes 1% and Radiant mining 0.75 as individual algorithms, you pay 1% on the EthB3 portion and 0.75% on the Radiant portion when dual mining. Usage is the same as described [here](https://github.com/Lolliedieb/lolMiner-releases/wiki/lolMiner-1.43-dual-and-split-mining-options) for Ethash.
- The algorithms ALEPH, ETHASHB3, IRONFISH, KARLSEN, NEXA and RADIANT now support **splitting the hashrate** analog to the options described [here](https://github.com/Lolliedieb/lolMiner-releases/wiki/lolMiner-1.43-dual-and-split-mining-options). Use --dualmode SPLIT-X to put X% of the hashrate to the 2nd connection given by --dualuser and --dualpool
- When using split mining, the hashrate will now be displayed as one solver as well in stats as in the API. This gives a more consistent representation.
- Reduced stale shares on algorithms ALEPH, ETHASHB3, IRONFISH, NEXA and RADIANT. 
- Windows: Updated **GUI tool**
- Added support for the **FishHash** algorithm (Future Ironfish PoW) for testing purposes. Use --algorithm FISHHASH-TEST to connect to a current Ironfish pool and run the shares just against the build in CPU verify. In this mode the difficulty is 1/256 of what the usual Blake3 difficulty would be. Use --algoriithm FISHHASH-TESTNET to connect to a real testnet pool and mine there including submitted shares. 

## Fixes

- Fixed a bug causing BEAM mining to crash on newer cuda versions.  
- Fixed a bug causing multiple algorithms to show hashrate in wrong place of statistics when running two algorithms on different GPUs.
- Fixed a bug with Rethereum mining not working on Cuda 11 drivers. 

## Note on Dual Mining and how to use an unknown algorithm:
As you might know you can use --dualmode <put and algorithm type here> according with --dualdevices to mine an algorithm that is completely different to the one provided by --algo. So e.g. if you use --algo ETHASHB3 --dualalgo KARLSEN --dualdevices ALL, you will actually only mine Heavyhash-Karlson to the pool and user given to --dualpool and --dualuser. Note that this entry is different from real dual mining, there you must use KARLSENDUAL to enable real dual mining both algorithms on one card. This difference is important, as the miner can run into buggy behavior if you use the wrong parameter entry for your purpose. 

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/1.78'>1.78</a>.</em>

**Comment by k0lenval** [comment]
Hello! I have a very limited hashrate on the Karlsen algorithm on Windows
